src/features/sarcophagi/components/Rewrap.tsx

```diff
@@ -30,29 +29,42 @@ export function Rewrap() {
   const rewrapIntervalSeconds = sarcophagus?.maximumRewrapInterval?.toNumber() ?? 0;
   const rewrapIntervalMs = rewrapIntervalSeconds * 1000;
 
-  function handleCustomDateChange(date: Date | null) {
-    setResurrectionTime(date);
+  function handleCustomDateChange(date: Date | null): void {
+    // Ensure that selected date is in the future
+    if (date && date.getTime() > Date.now()) {
+      setResurrectionTime(date);
+    }
   }
 
   const maximumResurectionDate = new Date(nowMs + rewrapIntervalMs);
-  const resurectionDateMs = maximumResurectionDate.getTime();
+  const resurrectionDateMs = maximumResurectionDate.getTime();
 
   const filterInvalidTime = (time: Date) => {
     const selectedDateMs = new Date(time).getTime();
-    return resurectionDateMs >= selectedDateMs && nowMs < selectedDateMs;
+    return resurrectionDateMs >= selectedDateMs && nowMs < selectedDateMs;
   };
 
   const resurrectionString = buildResurrectionDateString(
-    BigNumber.from(Math.trunc(resurectionDateMs / 1000))
+    BigNumber.from(Math.trunc(resurrectionDateMs / 1000))
   );
 
   // Sum up the digging fees from the archaeologists objects
-  const totalDiggingFeeBn = useMemo(() => {
+  const totalDiggingFeesPerSecBn = useMemo(() => {
     return archaeologists?.reduce((acc, archaeologist) => {
-      return acc.add(archaeologist.diggingFee);
+      return acc.add(archaeologist.diggingFeePerSecond);
     }, BigNumber.from(0));
   }, [archaeologists]);
-  const protocolFeeBn = totalDiggingFeeBn.mul(protocolFeeAmountInt).div(100);
+
+  const totalDiggingFeesBn = useMemo(() => {
+    if (!resurrectionTime) return BigNumber.from(0);
+    const resurrectionTimeSecBn = BigNumber.from(
+      Math.floor(new Date(resurrectionTime).getTime() / 1000)
+    );
+    const currentTimeSecBn = BigNumber.from(Math.floor(new Date().getTime() / 1000));
+    return totalDiggingFeesPerSecBn.mul(resurrectionTimeSecBn.sub(currentTimeSecBn));
```

Comment:
I _think_ you can get away with doing `.mul(<number> - <number>)` without having to convert `<number>`s to `BN`s first. Hopefully I'm remembering correctly

Reply:
There probably is no need to convert these values to big numbers. Now that I'm thinking about it your suggestion would make things a litter cleaner.